src/components/Nav/Navigation.tsx

```diff
@@ -99,41 +101,52 @@ class Navigation extends React.Component<PropsType, StateType> {
-            <PfContainerNavVertical>
-              <h2>Sorry, there was a problem. Try a refresh or navigate to a different page.</h2>
-            </PfContainerNavVertical>
-          )}
-        >
-          <Route path="/service-graph/:namespace" component={ServiceGraphRouteHandler} />
-          <Route path={servicesPath} component={ServiceListPage} />
-          <Route path={servicesJaegerPath} component={ServiceJaegerPage} />
-          <Route path="/namespaces/:namespace/services/:service" component={ServiceDetailsPage} />
-          <Route path={istioConfigPath} component={IstioConfigPage} />
-          <Route path="/namespaces/:namespace/istio/:objectType/:object" component={IstioConfigDetailsPage} />
-          <Redirect to={serviceGraphPath} />
-        </SwitchErrorBoundary>
+        {authenticated ? (
+          <div>
+            <VerticalNav
+              setControlledState={this.setControlledState}
+              activePath={this.state.selectedItem}
+              navCollapsed={this.state.navCollapsed}
+            >
+              <VerticalNav.Masthead title="Kiali">
+                <VerticalNav.Brand iconImg={pfLogo} titleImg={pfBrand} />
+                <VerticalNav.IconBar>
+                  <MessageCenter.Trigger />
+                  <HelpDropdown />
+                  <UserDropdown />
+                </VerticalNav.IconBar>
+                <MessageCenter drawerTitle="Message Center" />
+              </VerticalNav.Masthead>
+              <VerticalNav.Item title={serviceGraphTitle} iconClass="fa pficon-topology" onClick={this.navigateTo} />
+              <VerticalNav.Item title={servicesTitle} iconClass="fa pficon-service" onClick={this.navigateTo} />
+              <VerticalNav.Item title={istioConfigTitle} iconClass="fa pficon-template" onClick={this.navigateTo} />
+              <VerticalNav.Item title={servicesJaeger} iconClass="fa fa-paw" onClick={this.navigateTo} />
+            </VerticalNav>
+            <SwitchErrorBoundary
+              fallBackComponent={() => (
+                <PfContainerNavVertical>
+                  <h2>Sorry, there was a problem. Try a refresh or navigate to a different page.</h2>
+                </PfContainerNavVertical>
+              )}
+            >
+              <Route path="/service-graph/:namespace" component={ServiceGraphRouteHandler} />
+              <Route path={servicesPath} component={ServiceListPage} />
+              <Route path={servicesJaegerPath} component={ServiceJaegerPage} />
+              <Route path="/namespaces/:namespace/services/:service" component={ServiceDetailsPage} />
+              <Route path={istioConfigPath} component={IstioConfigPage} />
+              <Route path="/namespaces/:namespace/istio/:objectType/:object" component={IstioConfigDetailsPage} />
+              <Redirect to={serviceGraphPath} />
+            </SwitchErrorBoundary>
+          </div>
+        ) : (
+          <div>
+            <ConnectedLoginPage dispatch={null} />
```

Comment:
Perhaps I am wrong, I do not remember how ended the discussion about using null vs undefined.
I just see this one, but please, ignore it if it doesn't make sense.

Reply:
No .... you're right I remember some discussion about I'll change this